src/compiler/compiler.cc

```diff
@@ -797,23 +791,18 @@ ast::Unit* Pipeline::parse(Source* source) {
   return parser.parse_unit();
 }
 
-void Pipeline::do_with_lsp_selection_handler(const std::function<void (LspSelectionHandler*)>& callback) {
-  callback(null);
+void Pipeline::setup_lsp_selection_handler() {
```

Comment:
Now that there is an `Lsp` class, it's easier to just set the handler there, than to use this `do_with_lsp_selection_handler`.
The `Pipeline::run` now calls `setup_lsp_selection_handler` before it does a `resolve`.